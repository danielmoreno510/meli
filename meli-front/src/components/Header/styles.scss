@import '../../index';

.header-container {
  background-color: $aplication-color;
  height: 48px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-left: 160px;
  padding-right: 160px;
}

.header-logo {
  width: 44px;
  height: 32px;
  top: 8px;
  left: 10px;
  margin-right: 20px;
  cursor: pointer;
}

.header-form {
  flex: 1;
  display: flex;
  height: 32px;
}

.header-input {
  padding: 7px 60px 9px 15px;
  border-radius: 2px;
  background-color: $layout-color;
  border: 0 rgba(0, 0, 0, 0.2);
  width: calc(100% - 75px);
  color: #333;
  font-size: 16px;
  width: 100%;
  margin: 0;
  font-family: inherit;
  outline: 0;
}

.header-button {
  background-image: none;
  height: 32px;
  padding-top: 2px;
  cursor: pointer;
  border-radius: 0 2px 2px 0;
  width: 46px;
  border: 0 rgba(0, 0, 0, 0.2);
  background-color: $layout-color;
  top: 8px;
  right: 1px;
  outline: 0;
}

.header-icon{
  width: 28px;
  border-left: 1px solid #ddd;
}
