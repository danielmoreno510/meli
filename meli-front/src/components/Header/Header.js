import React, { useState, useEffect } from "react";
import { useHistory, useLocation } from "react-router-dom";
import queryString from "query-string";
import PropTypes from "prop-types";

import logo from "../../assets/images/logo.png";
import magnifying_glass from "../../assets/images/magnifying_glass.png";
import "./styles.scss";

const Header = ({ getProductList }) => {
  const location = useLocation();
  const value = queryString.parse(location.search);
  const history = useHistory();
  const [search, setSearch] = useState("");

  useEffect(() => {
    if (value.search) {
      setSearch(value.search);
      getProductList(value.search);
    }
  }, []);

  return (
    <div className="header-container">
      <img
        className="header-logo"
        src={logo}
        alt="logo"
        onClick={() => history.push(`/`)}
        data-testid='header-logo'
      />
      <div className="header-form">
        <input
          value={search}
          onChange={({ target }) => setSearch(target.value)}
          className="header-input"
          type="text"
          placeholder="Buscar productos, marcas y más…"
          data-testid='header-input'
        />
        <button
          data-testid='search-button'
          className="header-button"
          type="submit"
          onClick={() => {
            history.push(`/items?search=${search}`);
            getProductList(search);
          }}
        >
          <img
            className="header-icon"
            src={magnifying_glass}
            alt="magnifying glass"
          />
        </button>
      </div>
    </div>
  );
};

Header.propTypes = {
  getProductList: PropTypes.func,
};

export default Header;
