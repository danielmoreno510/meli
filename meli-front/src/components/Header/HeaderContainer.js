import Header from "./Header";
import { fetchProductList } from "../../state/actions";
import { StoreService } from "../../state/StoreService.js";

export const mapStateToProps = (state) => ({
});

export const mapDispatchToProps = (dispatch) => {
  return {
    getProductList: (search) => {
      dispatch(fetchProductList(search));
    }
  };
};
const HeaderContainer = StoreService.connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
export default HeaderContainer;
