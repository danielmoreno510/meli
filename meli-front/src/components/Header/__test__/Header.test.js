import React from "react";
import { shallow } from "enzyme";

import Header from "../Header";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useLocation: jest
      .fn()
      .mockReturnValue({ pathname: "/items", search: "?search=iphone%2012" }),
    useHistory: () => ({
      push: mockHistoryPush,
    }),
  };
});

describe("Header component", () => {
  const testProps = {
    getProductList: jest.fn(),
  };

  describe("WHEN the component is render", () => {
    const wrapper = shallow(<Header {...testProps} />);

    it("THEN should display Header component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("WHEN the Button is pressed", () => {
    const wrapper = shallow(<Header {...testProps} />);

    it("THEN getProductList should be called", () => {
      wrapper.find(`[data-testid="search-button"]`).simulate("click");
      expect(testProps.getProductList).toBeCalledTimes(1);
    });
  });

  describe("WHEN the Image is pressed", () => {
    const wrapper = shallow(<Header {...testProps} />);

    it("THEN push location should be called /", () => {
      wrapper.find(`[data-testid="header-logo"]`).simulate("click");
      expect(mockHistoryPush).toHaveBeenCalledWith('/');
    });
  });
});
