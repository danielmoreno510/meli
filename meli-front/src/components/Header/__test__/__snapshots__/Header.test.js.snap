// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component WHEN the component is render THEN should display Header component 1`] = `
<div
  className="header-container"
>
  <img
    alt="logo"
    className="header-logo"
    data-testid="header-logo"
    onClick={[Function]}
    src="logo.png"
  />
  <div
    className="header-form"
  >
    <input
      className="header-input"
      data-testid="header-input"
      onChange={[Function]}
      placeholder="Buscar productos, marcas y más…"
      type="text"
      value=""
    />
    <button
      className="header-button"
      data-testid="search-button"
      onClick={[Function]}
      type="submit"
    >
      <img
        alt="magnifying glass"
        className="header-icon"
        src="magnifying_glass.png"
      />
    </button>
  </div>
</div>
`;
